Minor cppcheck fixes

firefly2442 opened the pull request:
Hopefully it's ok to lump these together, if not let me know.  Found via [cppcheck](http://cppcheck.sourceforge.net/).

akien-mga commented on core/os/dir_access.h:
```diff
@@ -92,7 +92,7 @@ class DirAccess {
 
-	Error copy_dir(String p_from, String p_to, int chmod_flags = -1);
+	Error copy_dir(String p_from, String p_to, int p_chmod_flags = -1);
 	virtual Error copy(String p_from, String p_to, int chmod_flags = -1);
```
Would be good to do it here too for consistency.

akien-mga commented:
I think you could squash the commits together, they're all quite simple so they don't really need to be split out.

firefly2442 committed:
found via cppcheck:
remove code that will never run
make definition and declaration names for parameters match
change floats that were being set to bool values
remove pointer that is never used

firefly2442 commented:
Also adjusted the `copy` method per your suggestion and squashed.
